File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        count = defaultdict(int)

        for num in nums:
            count[num] += 1
        
        answer = 0
        for num in nums:
            if count[num-1] != 0:
                answer = max(answer,count[num] + count[num-1])
            if count[num+1] != 0:
                answer = max(answer, count[num] + count[num+1])
        
        return answer

File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> cnt;

        for (int num:nums){
            cnt[num]++;
        }

        int answer = 0;

        for(int num:nums){
            if(cnt.count(num+1)){
                answer = max(answer, cnt[num] + cnt[num+1]);
            }
        }

        return answer;
    }
};